frontend/src/lib/StaticGraphParser.ts

```diff
@@ -172,13 +198,66 @@ function buildTektonDag(graph: dagre.graphlib.Graph, template: any): void {
       ? 'cornsilk'
       : undefined;
 
-    graph.setNode(taskName, {
-      bgColor: bgColor,
-      height: Constants.NODE_HEIGHT,
-      info,
-      label: parseTaskDisplayName(task['taskSpec']) || label,
-      width: Constants.NODE_WIDTH,
-    });
+    if (task['taskSpec']) {
+      graph.setNode(taskName, {
+        bgColor: bgColor,
+        height: Constants.NODE_HEIGHT,
+        info,
+        label: parseTaskDisplayName(task['taskSpec']) || label,
+        width: Constants.NODE_WIDTH,
+      });
+    } else if (task['taskRef'] && task['taskRef']['kind'] === 'PipelineLoop') {
+      // handle the case of loop pipelines
+      graph.setNode(taskName, {
+        bgColor: bgColor,
+        height: Constants.NODE_HEIGHT,
+        info,
+        label: 'start-loop-' + loopNumber,
+        width: Constants.NODE_WIDTH,
+      });
+      const loopPipelineName = task['taskRef']['name'];
+      if (
+        pipeline['metadata'] &&
+        pipeline['metadata']['annotations'] &&
+        pipeline['metadata']['annotations']['tekton.dev/' + loopPipelineName]
```

Comment:
Do we have any sample pipelines using this annotation?

Reply:
Thanks @drewbutlerbb4 , yes, I will modify sdk part test examples for it, but seems I have to suspend this PR due to this logic should be removed after custom task could be defined as taskSpec